code/game/gamemodes/cm_self_destruct.dm

```diff
@@ -187,6 +187,287 @@ var/global/datum/authority/branch/evacuation/EvacuationAuthority //This is initi
+	var/obj/structure/machinery/lifeboat_bolt/id_panel/sister_panel
+	var/stage = LOCKED //Unlocked, Locked, Launching & Launched
+	var/aim_status	//Aiming, Aimed, Firing
+
+/obj/structure/machinery/lifeboat_bolt/target/get_examine_text()
+	. = ..()
+	if(sister_panel.primed)
+		. += SPAN_NOTICE("The control panel is <span class='bold'>primed</span>.\n<span class='bold'>Aim</span> then <span class='bold'>fire</span> to ignite the fuse and launch the lifeboat.")
+	else
+		. += SPAN_NOTICE("The control panel is <span class='bold'>deactivated</span>.\nPrime it using the other Control Panel.")
+
+/obj/structure/machinery/lifeboat_bolt/target/update_icon()
+	. = ..()
+	switch(stage)
+		if(LOCKED)
+			icon_state = "closed"
+		if(UNLOCKED)
+			if(sister_panel.primed)
+				icon_state = "red"
+			else
+				icon_state = "opened"
+		if(LAUNCHING)
+			icon_state = "flashing"
+		if(LAUNCHED)
+			icon_state = "released"
+
+
+/obj/structure/machinery/lifeboat_bolt/target/LateInitialize()
+	if(sister_panel) return
+
+	for(var/obj/structure/machinery/lifeboat_bolt/id_panel/M in machines)
+		if(M.id == id)
+			M.sister_target = src
+			sister_panel = M
+			break
+
+	if(!sister_panel) stat |= BROKEN
+
+//ANIMATION & SOUNDFX
+/obj/structure/machinery/lifeboat_bolt/target/proc/targeting_overlay()
+	overlays.Cut()
+	var/image/I = image('icons/effects/Targeted.dmi',"locking_movement")
+	I.pixel_x = 1
+	overlays += I
+	aim_status = AIMING
+	playsound(src,'sound/weapons/TargetOn.ogg', 35)
+
+/obj/structure/machinery/lifeboat_bolt/target/proc/targeted_overlay()
+	overlays.Cut()
+	var/image/I = image('icons/effects/Targeted.dmi',"locked_wide")
+	I.pixel_x = 1
+	overlays += I
+	aim_status = AIMED
+	playsound(src,'sound/weapons/TargetOn.ogg', 35)
+
+/obj/structure/machinery/lifeboat_bolt/target/proc/remove_target_overlay()
+	overlays.Cut()
+	aim_status = null
+	playsound(src,'sound/weapons/TargetOff.ogg', 35)
+
+/obj/structure/machinery/lifeboat_bolt/target/proc/unlock()
+	if(stage != LOCKED)
+		return
+	stage = UNLOCKED
+	flick("opening",src)
+	flick("opening_raising",sister_panel)
+	update_icon()
+	sister_panel.update_icon()
+
+//TIMED TO MATCH EXPLOSION, for best effect
+//Using Sleep is concerning, I can't remember why I didn't call Async on this
+/obj/structure/machinery/lifeboat_bolt/target/proc/detonate_animation()
+	stage = LAUNCHING
+	update_icon()
+	sister_panel.update_icon()
+	sleep(30)
+	flick("releasing",src)
+	stage = LAUNCHED
+	update_icon()
+	sleep(10)
+	sister_panel.update_icon()
+
+/obj/structure/machinery/lifeboat_bolt/target/proc/detonate()
+	if(stage >= LAUNCHING)
+		return
+
+	var/msg_log = "[key_name_admin(operator)] has launched the lifeboat [id]."
+	if(sister_panel.primed == PRIMED_TEMP) msg_log += "\n With Help from [sister_panel.operator]."
+	log_game(msg_log)
+	message_staff(msg_log)
+
+	operator.apply_effect(3,STUN)
+	shake_camera(operator, 15, 1, 2)
+
+	for(var/obj/docking_port/stationary/lifeboat_dock/lifeboat_dock in GLOB.lifeboat_almayer_docks)
```

Comment:
```suggestion
	for(var/obj/docking_port/stationary/lifeboat_dock/lifeboat_dock as anything in GLOB.lifeboat_almayer_docks)
```
i presume this list should never have anything that isn't a lifeboat dock anyway